### PR TITLE
CCIP-2573 - Handle TokenDeployer client creation when there is more than 1 router per chain

### PR DESCRIPTION
**Pull request by AnieeG**
## Solution

### CODE REVIEW

File: integration-tests/ccip-tests/testconfig/tomls/ccip-default.toml [hunk 1]
```diff
@@ -347,6 +351,10 @@ TimeoutForPriceUpdate = '15m' # Duration to wait for the price update to time-ou
 # Could be removed once the pipeline is completely removed.
 WithPipeline = false
 NoOfTokensPerChain = 2 # number of bridge tokens to be deployed per network; if MsgType = 'Token'/'DataWithToken'
+CCIPOwnerTokens = false # if true, the test will use deploy the tokens by the CCIPOwner, otherwise the tokens will be deployed by a non-owner account, only applicable for 1.5 pools and onwards
```
Comment (kalverra) [inline integration-tests/ccip-tests/testconfig/tomls/ccip-default.toml]: Thanks for adding this, though I think we should make it `true` by default. If you run with an older contract, the test should just fail.

**Review by kalverra — CHANGES_REQUESTED** [review]

File: integration-tests/ccip-tests/testconfig/tomls/ccip-default.toml (hunk 1 above)
Comment (AnieeG) [reply integration-tests/ccip-tests/testconfig/tomls/ccip-default.toml]: The default runs the test with latest contract, should it not be false in this case?
I can add one more comment mentioning that you need to change it to true , if you are running with pre-1.5 pools

File: integration-tests/ccip-tests/testconfig/tomls/scalability.toml [hunk 2]
```diff
@@ -91,6 +91,7 @@ TimeoutForPriceUpdate = '15m'
 NoOfTokensPerChain = 60
 NoOfTokensWithDynamicPrice = 15
 DynamicPriceUpdateInterval ='15s'
+CCIPOwnerTokens = true
```
Comment (AnieeG) [inline integration-tests/ccip-tests/testconfig/tomls/scalability.toml:94]: Here I have made it as true , as the contract version is 1.4

File: integration-tests/ccip-tests/testconfig/tomls/contract-version1.4.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[CCIP]
```
Comment (AnieeG) [inline integration-tests/ccip-tests/testconfig/tomls/contract-version1.4.toml:1]: Added this to ensure CI tests are running with previous versioned contracts as well

**Review by kalverra — APPROVED** [review]

**Review by jasonmci — APPROVED** [review]
approved

**Pull request merged**